src/Service/Parsers/ODataASTVisitor.cs

```diff
@@ -251,5 +257,34 @@ private static string GetFilterPredicateOperator(UnaryOperatorKind op)
                     throw new ArgumentException($"Uknown Predicate Operation of {op}");
             }
         }
+
+        /// <summary>
+        /// Helper method to populate the DbType for the property referenced in the OData filter.
+        /// Since the nodes are processed in a postorder fashion, the current BinaryOperatorNode has already been processed,
+        /// and the parameter is already created for the property.
+        /// We just need to populate the DbType.
+        /// </summary>
+        /// <param name="nodeIn">Binary operator node<</param>
+        private void PopulateDbTypeForProperty(BinaryOperatorNode nodeIn)
+        {
+            SingleValuePropertyAccessNode propertyNode = nodeIn.Left.GetType() == typeof(SingleValuePropertyAccessNode) ?
+                    (SingleValuePropertyAccessNode)nodeIn.Left : (SingleValuePropertyAccessNode)nodeIn.Right;
+            string? paramName = $"{BaseQueryStructure.PARAM_NAME_PREFIX}param{_struct.Counter.Current() - 1}";
+            _metadataProvider.TryGetBackingColumn(_struct.EntityName, propertyNode.Property.Name, out string? backingColumnName);
+            _struct.Parameters[paramName].DbType = _struct.GetUnderlyingSourceDefinition().Columns[backingColumnName!].DbType;
```

Comment:
- There is a mix of `TryGet` and direct access using `[index]`. 
- You're not handling the scenario when `TryGet` fails with `false`, which would result in a null reference exception with `Columns[backingColumnName!]`. The null forigiving operator doesn't look to be accurate here.

Reply:
since this function has the potential to throw exceptions, you may want to include that in the function comment. 

Would wrapping this logic in try/catch reduce the chances of failure bubbling exceptions into a 500 error returned to the client?

Reply:
TryGet won't return false here. If the columnName is non-existent, we would not reach this point in code.